frontend/public/components/catalog/catalog-item-details.jsx

```diff
@@ -11,13 +11,20 @@ import { SyncMarkdownView } from '../markdown-view';
 export class CatalogTileDetails extends React.Component {
   state = {
     plans: [],
+    markdown: '',
   };
 
   componentDidMount() {
-    const { obj, kind } = this.props.item;
+    const { obj, kind, markdownDescription } = this.props.item;
     if (kind === 'ClusterServiceClass') {
       this.getPlans(obj);
     }
+
+    if (_.isFunction(markdownDescription)) {
+      markdownDescription().then((md) => this.setState({ markdown: md }));
+    } else {
+      this.setState({ markdown: markdownDescription });
```

Comment:
When the markdown description is a string, this will trigger a rerender.

I suggest in the constructor to init the markdown state when the prop is a string.

Reply:
why do we not do the whole Markdown comparison/change in `catlog-items` file and simply pass on the string markdown value in props.item?

Reply:
we don't want to make a network call for the helm README file in this case until we need it. Otherwise we could end up doing many calls for nothing,